kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -125,12 +128,34 @@ private[kyuubi] class EngineRef private(conf: KyuubiConf, user: String, sessionI
       }
   }
 
-  private def create(zkClient: CuratorFramework): (String, Int) = tryWithLock(zkClient) {
-    // TODO: improve this after support engine pool. (KYUUBI #918)
-    var engineRef = getServerHost(zkClient, engineSpace)
-    // Get the engine address ahead if another process has succeeded
-    if (engineRef.nonEmpty) return engineRef.get
+  def getOrCreate(zkClient: CuratorFramework): (String, Int) = tryWithLock(zkClient) {
```

Comment:
Hmm, we need to wait for the lock every time..

Reply:
Yeah, I am thinking how to improve this.

Reply:
When support two more engines under same space, we‘ll face concurrency issues. Had tried serveral ways to improve the lock, but failed. Clients will wait for the lock only when connect to the kyuubi server, and these connections are long-time connections.